### PR TITLE
Split button template into two seperate button partials

### PR DESCRIPTION
**Pull request by jordythevulder**
Handy for when you want to overwrite the styling of the buttons

### CODE REVIEW

**Comment by royduin** [comment]
What if... it was a component? And you don't have to use the `@include` but use the component like `<x-...`. When the buttons are slots we're getting multiple options to change things:

- Default; just use the component
- Change just the buttons with 2 named slots
- Override the whole component

With the first 2 options you don't have to override a file; with the last option just 1. Maybe a better approach instead of splitting everything in multiple partials?

**Comment by jordythevulder** [comment]
Good one. I've extracted the file into a component so you can use the file however you'd like. I also kept the main subscribe-form.blade.php file and included the component inside of it to keep the backwards compatibility.

**Comment by royduin** [comment]
So the new partials can be removed now? Or you could choose the keep and use them within the component so you've even more options to change things. Missing the slot attribute merging btw.

**Comment by jordythevulder** [comment]
Yeah! i've kept the partials and included them. I've also indented the <graphql tags for a better structured view and added the attributes class merging on the buttons.

File: resources/views/components/subscribe-form/index.blade.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    :variables='{product_id: {{ $product_id }}, email: user.email}'
+    :callback="() => { window.app.$emit('alerts-updated') }"
+    :notify="{message: '@lang('We will let you know once this product is back in stock')', type: 'success'}"
+>
+    <div slot-scope="{ mutate }">
+        @if($subscribe->isNotEmpty())
+            {{ $subscribe }}
+        @else
+            @include('rapidez-product-alert::partials.subscribe')
+        @endif
```
Comment (royduin) [inline resources/views/components/subscribe-form/index.blade.php:17]: Would be nice to have a directive for this like;
```
@slot('subscribe')
    @include('rapidez-product-alert::partials.subscribe')
@endslot
```
And added within: https://github.com/rapidez/blade-directives, thoughts @indykoning?

File: resources/views/partials/login.blade.php [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<x-rapidez::button :href="route('account.login')" dusk="product-alert-login" :attributes="$login->attributes->class('flex items-center justify-center w-full')">
```
Comment (royduin) [inline resources/views/partials/login.blade.php]: Can't we do something like this?
```suggestion
<x-rapidez::button {{ $login->attributes->class('flex items-center justify-center w-full')->merge([
    'href' => route('account.login'),
    'dusk' => 'product-alert-login'.
]) }}>
```
Or this?
```suggestion
<x-rapidez::button {{ $login->attributes->merge([
    'class' => 'flex items-center justify-center w-full',
    'href' => route('account.login'),
    'dusk' => 'product-alert-login'.
]) }}>
```

Comment (jordythevulder) [reply resources/views/partials/login.blade.php]: Yes! i've added all the attributes in the merge!

Comment (royduin) [reply resources/views/partials/login.blade.php]: Nice, but is `:attributes` needed?

**Comment by royduin** [comment]
Merged; things left to check:

1. `:attributes`, can we do that differently?
2. `@slot()` or differently named directive

File: resources/views/components/subscribe-form/index.blade.php (hunk 1 above)
Comment (royduin) [reply resources/views/components/subscribe-form/index.blade.php:17]: Implementation: https://github.com/rapidez/blade-directives/pull/12, we could integrate this here in the future.

**Pull request merged**